9 layout: верстка главной страницы

rasulov1337 opened the pull request:
Мы выполнили верстку главной страницы

FireSpirit171 committed:
Add header layout

FireSpirit171 committed:
Add full header

rasulov1337 committed:
Add underline for currently selected page

FireSpirit171 committed:
Move header to components

rasulov1337 committed:
Add callbacks for headers

FireSpirit171 committed:
Add main photo container

rasulov1337 committed:
Fix position of search button

rasulov1337 committed:
Fix wrong search line + button positioning

rasulov1337 committed:
Add more spacing

FireSpirit171 committed:
Change mainPagePhoto css

rasulov1337 committed:
Fix typo

rasulov1337 committed:
Add callbacks

rasulov1337 committed:
Add code for advertisement cards

FireSpirit171 committed:
Add Filter component

FireSpirit171 committed:
Add primitive server

rasulov1337 committed:
Add .gitignore

rasulov1337 committed:
Update .gitignore

rasulov1337 committed:
Add handlebars support

rasulov1337 committed:
Move components into separate folders
Split CSS into more files

rasulov1337 committed:
Add props giving possible

rasulov1337 committed:
Add Ozon like effect for scrolling through the pictures of a card

rasulov1337 committed:
Change server port

rasulov1337 committed:
Add dependency to the package.json

FireSpirit171 committed:
Add authorization window

rasulov1337 committed:
Fix auth popup menu + its styles

rasulov1337 committed:
Fix error related to negative position of mouse relative to card

rasulov1337 committed:
Refactor AdCard code

rasulov1337 committed:
Update .gitignore

FireSpirit171 committed:
Move MainPhoto in components

FireSpirit171 committed:
Add registration

rasulov1337 committed:
Fix card on mouse hover bug
Add margin to main div
Add support for retrieving data from backend

rasulov1337 committed:
Move main page into separate file
Change names to better ones

rasulov1337 committed:
Fix logo being cropped

rasulov1337 committed:
Replace all unsafe innerHTML occurrences
Fix logo stretching

rasulov1337 committed:
Add Prettier & Eslint support
Reformat files using them

rasulov1337 committed:
Add comments for AdCard.js

rasulov1337 committed:
Add more comments

rasulov1337 committed:
Split CSS code into components
Fix logos being cropped
Fix styling

rasulov1337 committed:
Modify header's styles & code according to BEM

rasulov1337 committed:
Modify AdCard's styles & code according to BEM

rasulov1337 committed:
Add favicon

rasulov1337 committed:
Delete url prefix

rasulov1337 committed:
Add ajax + auth modules

FireSpirit171 committed:
Change CSS AuthPopup

FireSpirit171 committed:
Add basic validation

rasulov1337 committed:
Fix login/register cookie failure

rasulov1337 committed:
Add auth/reg support
Delete precompiled-templates.js from git

Nikita-hub000 commented on public/components/AdCard/AdCard.js:
```diff
@@ -0,0 +1,109 @@
+        this.currentImgIndex = 0
+        this.circles = []
+
+        if (parent === undefined || parent === null) {
```
лучше делать throw new Error

Nikita-hub000 commented on public/components/AdCard/AdCard.js:
```diff
@@ -0,0 +1,109 @@
+
+    render() {
+        // eslint-disable-next-line no-undef
+        const template = Handlebars.templates['AdCard.hbs']
```
убрать innerHtml
лучше через createElement и appendChild

rasulov1337 committed:
Update styles

Nikita-hub000 commented on public/components/AuthPopup/AuthPopup.js:
```diff
@@ -0,0 +1,228 @@
+                        minLen: 6,
+                        maxLen: 16,
+                    },
+                    password2: {
```
лучше passwordAgain

Nikita-hub000 commented on public/components/AuthPopup/AuthPopup.js:
```diff
@@ -0,0 +1,228 @@
+        login({
+            username: data['username'],
+            password: data['password'],
+        }).then((r) => {
```
нужен catch

Nikita-hub000 commented on public/components/AuthPopup/AuthPopup.js:
```diff
@@ -0,0 +1,228 @@
+                username: data['username'],
+                password: data['password'],
+                email: data['email'],
+            }).then((res) => {
```
нужен catch

Nikita-hub000 commented on public/components/MainPage/MainPage.js:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    render() {
+        fetch('http://localhost:8080/api/ads')
```
сюда напрашиваются envы

rasulov1337 committed:
Fix typo

Nikita-hub000 commented on public/components/MainPage/MainPage.js:
```diff
@@ -0,0 +1,42 @@
+                    const card = new AdCard(d, this.adsContainer)
+                    card.render()
+                }
+                this.pageContent.appendChild(this.adsContainer)
```
нужен catch 
и вообще лучше бы async await использовать 
читается легче

Nikita-hub000 commented on public/index.js:
```diff
@@ -0,0 +1,60 @@
+}
+
+/** Главная функция */
+;(async () => {
```
life это конечно хорошо, если вы будете в случае чего ответить на вопросы по ней)
вообще здесь можно использовать обычную функцию

rasulov1337 committed:
Add ips of server

Nikita-hub000 reviewed (changes requested):
Также не во всех js файлах есть use strict
и докиньте методам функций private public и тд

rasulov1337 committed:
Revert addresses back to localhost

FireSpirit171 committed:
Add full validation

rasulov1337 committed:
Move backend url into global var

rasulov1337 committed:
Update backend URL

FireSpirit171 committed:
Fix validation

rasulov1337 committed:
Update styles

rasulov1337 committed:
Fix auth error

FireSpirit171 committed:
Update Auth css

rasulov1337 committed:
Update AuthPopup styles

rasulov1337 committed:
Update AuthPopup styles

rasulov1337 committed:
Add more JSDOC comments

rasulov1337 committed:
Fix all remarks
Replace IIFE with arrow function
Refactor render() of MainPage.js
AuthPopup.js: password2 -> passwordAgain
AuthPopup.js: Add catch statements
AdCard.js: Throw error instead of logging error into console

rasulov1337 committed:
Update .gitignore

rasulov1337 committed:
Replace innerHTML with manual element creation
validationSigh -> validationSign

rasulov1337 committed:
Delete useless files

rasulov1337 committed:
Refactor code

rasulov1337 committed:
Refactor code

rasulov1337 committed:
Upgrade validation to the G tier

rasulov1337 committed:
Add wrong credentials text

rasulov1337 committed:
Add logout support

rasulov1337 committed:
Put template code into separate templateContainer

rasulov1337 committed:
Add existing account message handling
